scripts/check-test-determinism.py

```
@@ -0,0 +1,915 @@
+
+    total = len(positives) + len(negatives)
+    print(f"self-test OK: {len(positives)} positive + {len(negatives)} negative fixtures passed ({total} total)")
+    return 0
+
+
+# ---------------------------------------------------------------------------
+# CLI
+# ---------------------------------------------------------------------------
+
+
+def _resolve_roots(repo_root: pathlib.Path, roots: Optional[list[str]]) -> tuple[str, ...]:
```

Ruff (0.15.17)

[warning] Unused function argument: `repo_root`

(ARG001)